bosh-director/lib/bosh/director/deployment_plan/instance_plan.rb

```diff
@@ -269,7 +269,15 @@ def persist_current_spec
 
         def network_settings_changed?(old_network_settings, new_network_settings)
           return false if old_network_settings == {}
-          old_network_settings != new_network_settings
+          remove_dns_record_name_from_network_settings(old_network_settings) != new_network_settings
+        end
+
+        def remove_dns_record_name_from_network_settings(network_settings)
```

Comment:
let's use Bosh::Common::DeepCopy to make sure old_network_settings arent modified.